File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/net/codinsa2015/WeaponEnchantModelView.java
package net.codinsa2015;
import java.lang.*;
import java.util.ArrayList;
import java.io.BufferedReader;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.UnsupportedEncodingException;
import net.codinsa2015.StateAlterationModelView.*;
import java.util.ArrayList;


@SuppressWarnings("unused")
public class WeaponEnchantModelView
{


	// Obtient les altértions d'état appliquées à l'impact de l'attaque sur la cible.
	public ArrayList<StateAlterationModelView> OnHitEffects;
	// Obtient les altérations d'état appliquées lors de l'attaque sur le caster.
	public ArrayList<StateAlterationModelView> CastingEffects;
	// Obtient les effets passifs appliqués par l'enchantement.
	public ArrayList<StateAlterationModelView> PassiveEffects;
	public static WeaponEnchantModelView deserialize(BufferedReader input) throws UnsupportedEncodingException, IOException {
		WeaponEnchantModelView _obj =  new WeaponEnchantModelView();
		// OnHitEffects
		ArrayList<StateAlterationModelView> _obj_OnHitEffects = new ArrayList<StateAlterationModelView>();
		int _obj_OnHitEffects_count = Integer.valueOf(input.readLine());
		for(int _obj_OnHitEffects_i = 0; _obj_OnHitEffects_i < _obj_OnHitEffects_count; _obj_OnHitEffects_i++) {
			StateAlterationModelView _obj_OnHitEffects_e = StateAlterationModelView.deserialize(input);
			_obj_OnHitEffects.add((StateAlterationModelView)_obj_OnHitEffects_e);
		}
		_obj.OnHitEffects = _obj_OnHitEffects;
		// CastingEffects
		ArrayList<StateAlterationModelView> _obj_CastingEffects = new ArrayList<StateAlterationModelView>();
		int _obj_CastingEffects_count = Integer.valueOf(input.readLine());
		for(int _obj_CastingEffects_i = 0; _obj_CastingEffects_i < _obj_CastingEffects_count; _obj_CastingEffects_i++) {
			StateAlterationModelView _obj_CastingEffects_e = StateAlterationModelView.deserialize(input);
			_obj_CastingEffects.add((StateAlterationModelView)_obj_CastingEffects_e);
		}
		_obj.CastingEffects = _obj_CastingEffects;
		// PassiveEffects
		ArrayList<StateAlterationModelView> _obj_PassiveEffects = new ArrayList<StateAlterationModelView>();
		int _obj_PassiveEffects_count = Integer.valueOf(input.readLine());
		for(int _obj_PassiveEffects_i = 0; _obj_PassiveEffects_i < _obj_PassiveEffects_count; _obj_PassiveEffects_i++) {
			StateAlterationModelView _obj_PassiveEffects_e = StateAlterationModelView.deserialize(input);
			_obj_PassiveEffects.add((StateAlterationModelView)_obj_PassiveEffects_e);
		}
		_obj.PassiveEffects = _obj_PassiveEffects;
		return _obj;
	}

	public void serialize(OutputStreamWriter output) throws UnsupportedEncodingException, IOException {
		// OnHitEffects
		output.append(String.valueOf(this.OnHitEffects.size()) + "\n");
		for(int OnHitEffects_it = 0; OnHitEffects_it < this.OnHitEffects.size();OnHitEffects_it++) {
			this.OnHitEffects.get(OnHitEffects_it).serialize(output);
		}
		// CastingEffects
		output.append(String.valueOf(this.CastingEffects.size()) + "\n");
		for(int CastingEffects_it = 0; CastingEffects_it < this.CastingEffects.size();CastingEffects_it++) {
			this.CastingEffects.get(CastingEffects_it).serialize(output);
		}
		// PassiveEffects
		output.append(String.valueOf(this.PassiveEffects.size()) + "\n");
		for(int PassiveEffects_it = 0; PassiveEffects_it < this.PassiveEffects.size();PassiveEffects_it++) {
			this.PassiveEffects.get(PassiveEffects_it).serialize(output);
		}
	}

}

File: Codinsa2015/Codinsa2015/Server/Views/WeaponEnchantModelView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework;


namespace Codinsa2015.Views
{

	public class WeaponEnchantModelView
	{

		static Encoding BOMLESS_UTF8 = new UTF8Encoding(false);
	
		/// <summary>
		/// ID unique de l'enchantement.
		/// </summary>
		public int ID;	
		/// <summary>
		/// Obtient les altértions d'état appliquées à l'impact de l'attaque sur la cible.
		/// </summary>
		public List<StateAlterationModelView> OnHitEffects;	
		/// <summary>
		/// Obtient les altérations d'état appliquées lors de l'attaque sur le caster.
		/// </summary>
		public List<StateAlterationModelView> CastingEffects;	
		/// <summary>
		/// Obtient les effets passifs appliqués par l'enchantement.
		/// </summary>
		public List<StateAlterationModelView> PassiveEffects;	
		public WeaponEnchantModelView() {
			OnHitEffects = new List<StateAlterationModelView>();
			CastingEffects = new List<StateAlterationModelView>();
			PassiveEffects = new List<StateAlterationModelView>();
		}

		public static WeaponEnchantModelView Deserialize(System.IO.StreamReader input) {
			WeaponEnchantModelView _obj =  new WeaponEnchantModelView();
			// ID
			int _obj_ID = Int32.Parse(input.ReadLine());
			_obj.ID = (int)_obj_ID;
			// OnHitEffects
			List<StateAlterationModelView> _obj_OnHitEffects = new List<StateAlterationModelView>();
			int _obj_OnHitEffects_count = Int32.Parse(input.ReadLine());
			for(int _obj_OnHitEffects_i = 0; _obj_OnHitEffects_i < _obj_OnHitEffects_count; _obj_OnHitEffects_i++) {
				StateAlterationModelView _obj_OnHitEffects_e = StateAlterationModelView.Deserialize(input);
				_obj_OnHitEffects.Add((StateAlterationModelView)_obj_OnHitEffects_e);
			}
			_obj.OnHitEffects = (List<StateAlterationModelView>)_obj_OnHitEffects;
			// CastingEffects
			List<StateAlterationModelView> _obj_CastingEffects = new List<StateAlterationModelView>();
			int _obj_CastingEffects_count = Int32.Parse(input.ReadLine());
			for(int _obj_CastingEffects_i = 0; _obj_CastingEffects_i < _obj_CastingEffects_count; _obj_CastingEffects_i++) {
				StateAlterationModelView _obj_CastingEffects_e = StateAlterationModelView.Deserialize(input);
				_obj_CastingEffects.Add((StateAlterationModelView)_obj_CastingEffects_e);
			}
			_obj.CastingEffects = (List<StateAlterationModelView>)_obj_CastingEffects;
			// PassiveEffects
			List<StateAlterationModelView> _obj_PassiveEffects = new List<StateAlterationModelView>();
			int _obj_PassiveEffects_count = Int32.Parse(input.ReadLine());
			for(int _obj_PassiveEffects_i = 0; _obj_PassiveEffects_i < _obj_PassiveEffects_count; _obj_PassiveEffects_i++) {
				StateAlterationModelView _obj_PassiveEffects_e = StateAlterationModelView.Deserialize(input);
				_obj_PassiveEffects.Add((StateAlterationModelView)_obj_PassiveEffects_e);
			}
			_obj.PassiveEffects = (List<StateAlterationModelView>)_obj_PassiveEffects;
			return _obj;
		}

		public void Serialize(System.IO.StreamWriter output) {
			// ID
			output.WriteLine(((int)this.ID).ToString());
			// OnHitEffects
			output.WriteLine(this.OnHitEffects.Count.ToString());
			for(int OnHitEffects_it = 0; OnHitEffects_it < this.OnHitEffects.Count;OnHitEffects_it++) {
				this.OnHitEffects[OnHitEffects_it].Serialize(output);
			}
			// CastingEffects
			output.WriteLine(this.CastingEffects.Count.ToString());
			for(int CastingEffects_it = 0; CastingEffects_it < this.CastingEffects.Count;CastingEffects_it++) {
				this.CastingEffects[CastingEffects_it].Serialize(output);
			}
			// PassiveEffects
			output.WriteLine(this.PassiveEffects.Count.ToString());
			for(int PassiveEffects_it = 0; PassiveEffects_it < this.PassiveEffects.Count;PassiveEffects_it++) {
				this.PassiveEffects[PassiveEffects_it].Serialize(output);
			}
		}

	}
}
